deploy/backup/copy-backup.sh

```diff
@@ -68,58 +76,84 @@ enable_logging() {
-        usage
-    fi
+check_input_namespace() {
+    local namespace=${1}
 
-    if [ ! -e "$dest_dir" ]; then
-        mkdir -p "$dest_dir"
+    if [ -n "$namespace" ]; then
+        ctrl="$ctrl -n $namespace"
     fi
+}
 
-    if [ "${backup_dest:0:4}" = "pvc/" ]; then
-        if ! $ctrl get "$backup_dest" 1>/dev/null; then
-            printf "[ERROR] '%s' PVC doesn't exists.\n\n" "$backup_dest"
-            usage
-        fi
-    elif [ "${backup_dest:0:5}" = "s3://" ]; then
-        env -i $CREDENTIALS $xbcloud get ${backup_dest} xtrabackup_info 1>/dev/null
-    else
-        usage
-    fi
+check_input_destination() {
+  local backup_dest=$1
+  local dest_dir=$2
+
+  if [ -z "$backup_dest" ] || [ -z "$dest_dir" ]; then
+      usage
+  fi
+
+  if [ ! -e "$dest_dir" ]; then
+      mkdir -p "$dest_dir"
+  fi
+
+  if [ "${backup_dest:0:4}" = "pvc/" ]; then
+      if ! $ctrl get "$backup_dest" 1>/dev/null; then
+          printf "[ERROR] '%s' PVC doesn't exists.\n\n" "$backup_dest"
+          usage
+      fi
+  elif [ "${backup_dest:0:5}" = "s3://" ]; then
+      env -i $CREDENTIALS $xbcloud get ${backup_dest} xtrabackup_info 1>/dev/null
+  else
+      echo "Can't find $backup_dest backup"
+      usage
+  fi
+
+  if [ ! -d "$dest_dir" ]; then
+      printf "[ERROR] '%s' is not local directory.\n\n" "$dest_dir"
+      usage
+  fi
 
-    if [ ! -d "$dest_dir" ]; then
-        printf "[ERROR] '%s' is not local directory.\n\n" "$dest_dir"
-        usage
-    fi
 }
 
 start_tmp_pod() {
     local backup_pvc=$1
 
-    $ctrl delete pod/backup-access 2>/dev/null || :
-    cat - <<-EOF | $ctrl apply -f -
-		apiVersion: v1
-		kind: Pod
-		metadata:
-		  name: backup-access
-		spec:
-		      containers:
-		      - name: xtrabackup
-		        image: percona/percona-xtradb-cluster-operator:0.3.0-backup
-		        volumeMounts:
-		        - name: backup
-		          mountPath: /backup
-		      restartPolicy: Never
-		      volumes:
-		      - name: backup
-		        persistentVolumeClaim:
-		          claimName: ${backup_pvc#pvc/}
-	EOF
+    if [ -n "$namespace" ]; then
+      $ctrl delete pod/backup-access 2>/dev/null || echo "apiVersion: v1
+kind: Pod
+metadata:
+  name: backup-access
+  namespace: $namespace
+spec:
+  containers:
+  - name: xtrabackup
+    image: percona/percona-xtradb-cluster-operator:0.3.0-backup
+    volumeMounts:
+    - name: backup
+      mountPath: /backup
+  restartPolicy: Never
+  volumes:
+  - name: backup
+    persistentVolumeClaim:
+      claimName: ${backup_pvc#pvc/}" | $ctrl apply -f -
+    else
+      $ctrl delete pod/backup-access 2>/dev/null || echo "apiVersion: v1
+kind: Pod
+metadata:
+  name: backup-access
+spec:
+  containers:
+  - name: xtrabackup
+    image: percona/percona-xtradb-cluster-operator:0.3.0-backup
+    volumeMounts:
+    - name: backup
+      mountPath: /backup
+  restartPolicy: Never
+  volumes:
+  - name: backup
+    persistentVolumeClaim:
+      claimName: ${backup_pvc#pvc/}" | $ctrl apply -f -
```

Comment:
it is simple to lose `$ctrl apply -f -` command in the new syntax.
please keep using "bash here-doc" syntax

Reply:
submitted a commit to sort out this